### ISSUE TITLE
Why did I get the hash incorrectly?

### ISSUE THREAD

**Issue by zzyandzzy** [dampcake/bencode#5]
**java env**
```cmd
java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)
```

**critical code**
```java
/**
 * torrentFile see torrent.zip
 */
byte[] torrentData = FileUtils.readFileToByteArray(new File(torrentFile))
Bencode bencode = new Bencode();
Map<String, Object> torrentMap = bencode.decode(torrentData, Type.DICTIONARY);
Map<String, Object> info = (Map<String, Object>) torrentMap.get("info");
byte[] hashByte = hash(bencode.encode(info));
System.out.println(bytesToHex(hashByte));

public static String bytesToHex(byte[] bytes) {
        char[] hexChars = new char[bytes.length * 2];
        for (int j = 0; j < bytes.length; j++) {
            int v = bytes[j] & 0xFF;
            hexChars[j * 2] = HEX_SYMBOLS[v >>> 4];
            hexChars[j * 2 + 1] = HEX_SYMBOLS[v & 0x0F];
        }
        return new String(hexChars);
    }
```
**The correct hash is:**
 ```C8AE498D84E1ABC0EE9A72046E31E82B6A5C8C74```
**but i get hash is**
```D5FFE89589B9BACB195C2771C3FC328BF3880623```

[torrent.zip](https://github.com/dampcake/bencode/files/3116650/torrent.zip)

**thx, - ̗̀(๑ᵔ⌔ᵔ๑)**

**Comment by dampcake** [comment]
Looks like this is an issue with trying to make everything a `String` to make it easier to work with. I'll see if I can come up with a solution.

**Comment by dampcake** [comment]
I have just pushed version 1.3 to central which has a fix for this.
Using `new Bencode(true)` will now cause this code to produce the correct hash.

Note that using `new Bencode(true)` will make any values in `List`s or `Map`s returned by the decoder to `ByteBuffer`s. The keys for the `Map`s will still be coerced to `String`s as that is the spec for Bencode.

I'll get release notes written up for this soon and update the README. The JavaDocs have already been updated with the newness.
